src/SentryFullStory.ts

```diff
@@ -3,6 +3,7 @@ import { Event, EventHint } from '@sentry/types';
 import * as FullStory from '@fullstory/browser';
```

Comment:
Trying to get context on this, is the idea to let the user provide their own client as an option (which would be coming from Segment)? Even if they were to provide it, this package still requires `@fullstory/browser`, which is what #30 is concerning.

If you help me understand the benefit of this feature that'd be really helpful, I might just be misunderstanding! Thanks!

Reply:
@leeandher correct, in both observations.

I am trying to avoid breaking changes, but ideally.

A) It would be much better to remove the fullstory client, and expect people to pass it `always`.

Or much better, and I am inclined to this one

B) What would be the easiest way we can extract most of the code from this package that would allow people to create `sentry-segment`, `sentry-fullstory` (from scratch)?

Right now they are multiple things happening around the Fullstory itself (if you think about it), that some people (like me) feel the need to extend this one because of fear of maintaining the other code and things could break and requires them to be on top of Sentry SDK (except that Sentry owns these stuff right now).

So the simpler we make it for others to take advantage of such boilerplate code, the easier is to justify to them that they can put a tiny JS in their codebase and own it.

Does it make sense?

Reply:
Hey, hoping in this convo as a maintainer of https://github.com/getsentry/sentry-javascript - do you mind condensing some of these thoughts and opening an issue in the js sdk monorepo so we can track this across the JS sdks?

Reply:
Right on @AbhiPrasad let me put some thoughts there

Reply:
https://github.com/getsentry/sentry-javascript/issues/4272 @AbhiPrasad did my best there, not sure how much I should document, but I am interested in continuing the contribution however I can.

Reply:
I left some comments with some questions, @AbhiPrasad may help to understand from SDK development some concerns of confusing things.

Reply:
It’s on the docket to review on with the other maintainers on monday, appreciate the assistance!

We are also especially interested in these topics as we are planning for a major version bump soon- so maybe we can make this easier that way. https://github.com/getsentry/sentry-javascript/issues/4240